src/Microsoft.ML.StaticPipe/EvaluatorStaticExtensions.cs

```diff
@@ -50,6 +51,44 @@ public static class EvaluatorStaticExtensions
             return eval.Evaluate(data.AsDynamic, labelName, scoreName, probName, predName);
         }
 
+        /// <summary>
+        /// Evaluates scored binary classification data and generates precision recall curve data.
+        /// </summary>
+        /// <typeparam name="T">The shape type for the input data.</typeparam>
+        /// <param name="catalog">The binary classification catalog.</param>
+        /// <param name="data">The data to evaluate.</param>
+        /// <param name="label">The index delegate for the label column.</param>
+        /// <param name="pred">The index delegate for columns from calibrated prediction of a binary classifier.
+        /// Under typical scenarios, this will just be the same tuple of results returned from the trainer.</param>
+        /// <param name="prCurve">The generated precision recall curve data. Up to 100000 of samples are used for p/r curve generation.</param>
+        /// <returns>The evaluation results for these calibrated outputs.</returns>
+        public static CalibratedBinaryClassificationMetrics EvaluateWithPRCurve<T>(
+            this BinaryClassificationCatalog catalog,
+            DataView<T> data,
+            Func<T, Scalar<bool>> label,
+            Func<T, (Scalar<float> score, Scalar<float> probability, Scalar<bool> predictedLabel)> pred,
+            out IEnumerable<PrecisionRecallMetrics> prCurve)
```

Comment:
>prCurve [](start = 52, length = 7)

find the out param awkward. Can we extend the CalibratedBinaryClassificationMetrics instead?

Reply:
this would mean unsealing CalibratedBinaryClassificationMetrics class, its sealed currently

---
In reply to: [268283796](https://github.com/dotnet/machinelearning/pull/3039#discussion_r268283796) [](ancestors = 268283796)